Cargo.toml

```diff
@@ -1,21 +1,16 @@
 [package]
 name = "atomic-pool"
-version = "1.0.1"
+version = "1.0.2"
 authors = ["Dario Nieuwenhuis <dirbaio@dirbaio.net>"]
 description = "Statically allocated pool providing a std-like Box."
 repository = "https://github.com/embassy-rs/atomic-pool"
 edition = "2021"
 readme = "README.md"
 license = "MIT OR Apache-2.0"
-categories = [
-    "embedded",
-    "no-std",
-    "concurrency",
-    "memory-management",
-]
+categories = ["embedded", "no-std", "concurrency", "memory-management"]
 
 [dependencies]
-atomic-polyfill = "1.0"
 as-slice-01 = { package = "as-slice", version = "0.1.5" }
 as-slice-02 = { package = "as-slice", version = "0.2.1" }
+portable-atomic = { version = "1.7.0", features = ["critical-section"] }
```

Comment:
do not unconditionally enable `critical-section`, the user might want to use `unsafe-assume-single-core` instead which is slightly faster in single-core chips.

I'd add two features that forward to portable-atomic:

```toml
[features]
portable-atomic-critical-section = ["portable-atomic/critical-section"]
portable-atomic-unsafe-assume-single-core = ["portable-atomic/unsafe-assume-single-core"]